app/src/app/components/contents/downloads/FileRow.tsx

```diff
@@ -0,0 +1,51 @@
+import { ChartColumn, File, Presentation, TableProperties } from "lucide-react";
+import { bytesToSize } from "../../../../helpers";
+import { PacketMetadata } from "../../../../types";
+import DownloadButton from "./DownloadButton";
+
+interface FileRowProps {
+  path: string;
+  packet: PacketMetadata;
+  sharedResource?: boolean;
+}
+
+export default function FileRow({ path, packet, sharedResource }: FileRowProps) {
+  const extension = path.split(".").pop();
+  const file = packet.files.filter((file) => {
+    return file.path === path.replace("//", "/");
+  })[0];
+  const fileName = path.split("/").pop();
+
+  return (file &&
+    <div className="p-2 flex justify-between">
+      <div className="flex items-center truncate">
+        <span className="min-w-fit">
+          {extension && (() => {
```

Comment:
another component for this seems appropriate rather than the anonymous function

Reply:
Or how about [this](https://github.com/mrc-ide/packit/pull/151/commits/f674f0213f31596a60b8a5c47a9dc80b3928b2b0)?

Reply:
just saw and is alot nicer.. but `getFileIcon` is actually a jsx function but its being used like a regular function? so maybe best to just rename to `FileIcon` and extract out and use like `<Fileicon extension={extension}/>